### PR TITLE
config: make semaphore ci commands easy to run on local

### PR DESCRIPTION
**Pull request by pbludov**
From https://github.com/checkstyle/checkstyle/pull/8920#discussion_r515509604

> one day this execution will fail in CI in PR, contributor might not be very experienced in CI to figureout how to reproduce it on local, so we will share with him link to single line in config for simple copy and paste in terminal, and it works.

### CODE REVIEW

**Review by romani — APPROVED** [review]
good step forward but, now we all forced to build project all the time again and again if we want to execute some "no-error-xxxx".
When I dealing with CI problems build, I always benefit from skipping chekcstyle build.  

what if we do 

`mvn -e clean install -Pno-validations && .ci/validation.sh all-sevntu-checks`

or keep:
```
          commands:
            - mvn -e install -Pno-validations
```

@nmancus1 , please share your opinion.

**Comment by nmancus1** [comment]
>what if we do

mvn -e clean install -Pno-validations && .ci/validation.sh all-sevntu-checks

This is a good idea; I always comment out the lines to build Checkstyle when working on CI builds so that I don't have to keep building Checkstyle over and over again on my machine.  For new contributors, this keeps it easy to point them to a single line to copy, paste, and run locally as Roman mentioned in https://github.com/checkstyle/checkstyle/pull/8920#discussion_r515509604.  So, this is a good approach for both new contributors and those that are familiar with Checkstyle.

File: .semaphore/semaphore.yml
Comment (nmancus1) [inline .semaphore/semaphore.yml:46]: `check-missing-pitests` was assigned priority 10 originally, and `no-error-pmd` was assigned 80. Now, priority assignment will be automatic from semaphore. Out of all these tests, `no-error-pmd` is the most frequently failed one for me, so I think that it made sense for it to have higher priority so that the build fails faster and not waste time/resources. I am not sure how many builds we can run in parallel, and if this even matters, but I thought it would be good to mention.

**Comment by romani** [comment]
we might want to do 
`./.ci/prep-build.sh && .ci/validation.sh all-sevntu-checks`
vs
`mvn -e clean install -Pno-validations && .ci/validation.sh all-sevntu-checks`

to keep command shorter(less scary)

**Comment by nmancus1** [comment]
>`./.ci/prep-build.sh`

Would this script only consist of 'mvn -e clean install -Pno-validations'? Seems like it would obfuscate what we're doing a bit.  I think it's good to be clear that we are building Checkstyle, then running the validation.

**Comment by romani** [comment]
agree, I am ok with any approach, just tried to think loud.

while comamnd is less then 100 ... it is ok to keep all as is (no obfuscation)

File: .semaphore/semaphore.yml
Comment (pbludov) [reply .semaphore/semaphore.yml:46]: Makes sense. I'll make two jobs: one with high priority, another with default priority

**Comment by pbludov** [comment]
> mvn -e clean install -Pno-validations && .ci/validation.sh all-sevntu-checks

I agree. We show the full command to the users, and they decide whether to execute it as is or only part of it.

**Review by nmancus1 — APPROVED** [review]
I am ok to merge once CI is green.

File: .semaphore/semaphore.yml
Comment (romani) [inline .semaphore/semaphore.yml:74]: put fast in front... it will also group them a better  to read way
make it like:
```
                - .ci/validation.sh all-sevntu-checks
                - .ci/validation.sh check-missing-pitests
                - .ci/validation.sh eclipse-static-analysis
                - mvn -e clean install -Pno-validations
                    && .ci/no-exception-test.sh guava-with-google-checks
                - mvn -e clean install -Pno-validations
                    && .ci/no-exception-test.sh guava-with-sun-checks
                - mvn -e clean install -Pno-validations
                    && .ci/validation.sh no-violation-test-josm
```

**Review by romani — APPROVED** [review]
ok to merge , with minor todo before merge:

File: .semaphore/semaphore.yml
Comment (pbludov) [reply .semaphore/semaphore.yml:74]: done

**Pull request merged**
